src/http.js

```diff
@@ -1,13 +1,37 @@
 //------------------------------------------------------------------------------
```

Comment:
Yes, it's moving that direction. Specifically the ability to pass in params vs bodyParameters comes from there, although they name them differently. And just writing them down now, I realize there is a difference in the spelling of "parameters" between the two. I think I will unify those.